src/wfs/src/main/java/org/geoserver/wfs/request/Delete.java

```diff
@@ -102,4 +97,20 @@ protected void eAddForDelete(EObject obj, String property, Filter newFilter) {
             ((DeleteTypeImpl) obj).setFilter(result);
         }
     }
+
+    private List<Filter> combineFilters(Filter newFilter, Filter currentFilter) {
+        List<Filter> filters = new ArrayList<>();
+        if (newFilter != null) {
+            filters.add(newFilter);
+        }
+
+        if (currentFilter != null) {
+            if (currentFilter instanceof BinaryLogicOperator) {
+                filters.addAll(((BinaryLogicOperator) currentFilter).getChildren());
+            } else {
+                filters.add(currentFilter);
+            }
```

Comment:
Check seems too broad... the filters in the list are re-combined with or, so it should be checking for a "or" filter, rather than BinaryLogicOperator (which covers And as well):

![image](https://user-images.githubusercontent.com/325433/219417958-545f99bd-21ef-40f3-ab9f-0095f547b58a.png)